ui/plugins/pinboard.js

```diff
@@ -132,6 +132,18 @@ treeherder.controller('PinboardCtrl', [
             }
         };
 
+        // If the pasted data is (or looks like) a 12 or 40 char SHA,
+        // or if the pasted data is an hg.m.o url, automatically select
+        // the "fixed by commit" classification type
+        $scope.pasteSHA = function (evt) {
+            var pastedData = evt.originalEvent.clipboardData.getData('text');
+            var re1 = pastedData.search(/^[0-9a-f]{12}$/) >= 0;
+            var re2 = pastedData.search(/^[0-9a-f]{40}$/) >= 0;
+            if (re1 || re2 || pastedData.includes("hg.mozilla.org")) {
```

Comment:
The two regexes can be combined partially, plus use `\d` for 0-9. Also might be worth supporting lengths inbetween 12 and 40, which also simplifies the regex.

eg something like:
`/^[a-f\d]{12,40}$/`

I would also use `.test()` rather than `.search()`.

It would also be great to factor out the "contains a backout commit" functionality (not the `failure_classification_id` part) and add a unit test for it.